code/modules/mob/living/simple_animal/hostile/wolf.dm

```diff
@@ -0,0 +1,405 @@
+	harm_intent_damage = 8
+	melee_damage_lower = 15
+	melee_damage_upper = 20
+	attacktext = "bites"
+	attack_sound = 'sound/weapons/bite.ogg'
+
+	faction = "wolf"
+	attack_same = 1 //Handled more in CanAttack
+	minbodytemp = 200
+
+	var/alert = 0 //Listening out for pointings from the pack alpha
+	var/aggressive = AGGNO
+	var/anger_chance = 30
+	var/mob/living/pack_alpha //Who they will never attack, and if human, will listen to commands
+	var/alpha_stance = ALPHANONE //What the alpha may want them to do
+	var/atom/alpha_target //whomever the alpha has specified to attack
+	var/mob/alpha_challenger //Whomever is challenging the alpha
+	var/hunger_status //Scales off nutrition. 400+ = well fed, 300 hungry, 200 vhungry, 100 starving
+	var/alpha_challenge //Used only by pack alphas, used for duels
+	var/obj/effect/decal/point/point_last //Stores the last point we saw
+
+/mob/living/simple_animal/hostile/wolf/alpha
+	name = "wolf alpha"
+
+/mob/living/simple_animal/hostile/wolf/alpha/New()
+	..()
+	pack_alpha = src
+	name += " ([rand(0,999)])"
+	spawn(5)
+		var/list/can_see = view(src, vision_range)
+		for(var/mob/living/simple_animal/hostile/wolf/potential_pack in can_see)
+			if(potential_pack.isDead())
+				continue //He ded
+			if(potential_pack == src)
+				continue //That's us
+			if(!potential_pack.pack_alpha)
+				potential_pack.pack_alpha = src
+			else
+				if(istype(potential_pack.pack_alpha, /mob/living/simple_animal/hostile/wolf))
+					var/mob/living/simple_animal/hostile/wolf/alpha_challenger = potential_pack.pack_alpha
+					alpha_challenger.challenge(src)
+					challenge(alpha_challenger)
+
+/mob/living/simple_animal/hostile/wolf/CanAttack(var/atom/the_target)
+	//WE DON'T ATTACK INVULNERABLE MOBS (such as etheral jaunting mobs, or passengers of the adminbus)
+	if(the_target.flags & INVULNERABLE)
+		return 0
+	if(istype(the_target, /mob/dead/observer))
+		return 0 //Stop eating ghosts!
+	if((the_target == alpha_target && alpha_stance == ALPHAATTACK) || the_target == alpha_challenger)//Who the alpha has specified
+		return the_target //RIP sanity, but alpha above all
+	if(aggressive == AGGNO)
+		if(the_target == pack_alpha && alpha_stance == ALPHAFOLLOW)
+			return the_target
+		return 0
+	if(ismob(the_target))
+		var/mob/living/L = the_target
+		if(L == pack_alpha)
+			if(alpha_stance == ALPHAFOLLOW)
+				return the_target //Handled more in AttackingTarget
+			else
+				return 0
+		if(istype(L, /mob/living/simple_animal/hostile/wolf))
+			var/mob/living/simple_animal/hostile/wolf/potential_pack = L
+			if(potential_pack.pack_alpha == pack_alpha) //Never eat a packmate
+				return 0
+			if(aggressive != AGGALL)//Not unless we're hungry
+				return 0
+			else
+				return the_target
+		if(L.faction == src.faction)
+			if(aggressive != AGGALL)//Not unless we're hungry
+				return 0
+			else
+				return the_target
+		if(L.isDead())
+			if(hunger_status < WELLFED)
+				return the_target
+			else
+				return 0
+	return ..(the_target)
```

Comment:
..() keeps the same args automatically is what he means

Yes, the code is full of shitty code, we know, it doesn't excuse adding more.